### PR TITLE
docs #594 Add section about Composing Directives Java

### PR DESCRIPTION
**Pull request by jlprat**
Issue: #594
Add section explaining the different alternatives to compose routes using the Java DSL.
* Example using `orElse`
* Example combining directives in own method
* Example using `route`
* Example using `roture` with several alternatives

### CODE REVIEW

**Comment by jlprat** [comment]
PLS BUILD

File: docs/src/main/paradox/java/http/routing-dsl/directives/index.md [hunk 1]
```diff
@@ -83,4 +83,38 @@ transformations, both (or either) on the request and on the response side.
 
 ## Composing Directives
 
-TODO rewrite for Java API
```
Comment (ktoso) [inline docs/src/main/paradox/java/http/routing-dsl/directives/index.md]: wow that's embarrassing that we had a TODO left like this...
Thanks a ton @jlprat !

**Comment by ktoso** [comment]
LGTM, thanks a lot!

File: docs/src/main/paradox/java/http/routing-dsl/directives/index.md (hunk 1 above)
Comment (jlprat) [inline docs/src/main/paradox/java/http/routing-dsl/directives/index.md]: Well... at least it was an open and honest TODO :wink:

**Pull request merged**